diff --git a/config.py b/config.py
index 2a8ad3a..f5b646e 100644
--- a/config.py
+++ b/config.py
@@ -108,7 +108,7 @@ def build_interpreter(define_macros, include_dirs, libraries, library_dirs,
     # XXX some of this is duplicated in setup.py!  Why do the same thing twice?
     cfiles.sort()
 
-    sources = ['c/bc.c', 'c/localized_functions.c', 'c/mpi.c', 'c/_gpaw.c',
+    sources = ['c/bc.c', 'c/mpi.c', 'c/_gpaw.c',
                'c/operators.c', 'c/woperators.c', 'c/transformers.c',
                'c/elpa.c',
                'c/blacs.c', 'c/utilities.c', 'c/xc/libvdwxc.c']
