diff --git a/gpaw/test/parallel/pblas.py b/gpaw/test/parallel/pblas.py
index 3ead06d..e920969 100644
--- a/gpaw/test/parallel/pblas.py
+++ b/gpaw/test/parallel/pblas.py
@@ -55,7 +55,7 @@ def main(M=160, N=120, K=140, seed=42, mprocs=2, nprocs=2, dtype=float):
     # HEC = HEA * B
     HEA0 = gen.rand(*globHEC.shape) + epsilon * gen.rand(*globHEC.shape)
     if world.rank == 0:
-        HEA0 = HEA0 + HEA0.T.conjugate() # Make H0 hermitean
+        HEA0 += HEA0.T.conjugate() # Make H0 hermitean
     
     # Local result matrices
     Y0 = globY.empty(dtype=dtype)
